test/proxy.test.ts

```diff
@@ -105,7 +105,7 @@ describe.skip('Proxy Module TLS Certificate Loading', () => {
     process.env.NODE_ENV = 'test';
     process.env.GIT_PROXY_HTTPS_SERVER_PORT = '8443';
 
```

Comment:
I did some testing here while trying to remove the `.skip`, and although I couldn't find a proper fix, I did find a way to sweep the errors under the rug:

```suggestion

     vi.doMock('../src/config/env', () => ({
      serverConfig: {
        GIT_PROXY_SERVER_PORT: '8001',
        GIT_PROXY_HTTPS_SERVER_PORT: '8443',
      },
    }));
```

We probably don't want to do this - but it shows that the problem comes from some `Proxy` instance somewhere not closing properly...